src/wp-includes/class-wp-block.php

```diff
@@ -656,6 +621,63 @@ public function render( $options = array() ) {
 			$root_interactive_block = null;
 		}
 
+		/*
+		 * Proceed with enqueueing scripts and styles only if the block rendered any tags. A block may commonly not
+		 * render anything. Some examples:
+		 *
+		 *  - Featured Image block on a post without a featured image.
+		 *  - Comments block on a post with comments disabled.
+		 *  - An arbitrary block which has a `render_block` filter that returns an empty string (or an HTML comment).
+		 *
+		 * In all these cases, adding scripts and styles will be a waste since they will not be used on the page.
+		 */
+		$processor = new WP_HTML_Tag_Processor( $block_content );
```

Comment:
Is this worth to bypass the tag processor if possible?

```suggestion
		if ( empty( trim( $block_content ) ) ) {
			$enqueue = false;
		} else {
			$processor = new WP_HTML_Tag_Processor( $block_content );
			$enqueue = $processor->next_tag();
		}
```

plus the appropriate change to filter below.

Reply:
Good. Committed in de22ae9b4f528b9c082c21e018dc156449550a38.